src/utils/create-utils.ts

```diff
@@ -114,7 +114,12 @@ export const createComponent = (
   };
 
   const resource =
-    verb === 'update' ? merge({}, originalComponent || {}, newComponent) : newComponent;
+    verb === 'update'
+      ? {
+          ...merge({}, originalComponent || {}, newComponent),
+          spec: newComponent.spec,
```

Comment:
@karthikjeeyar This will completely replace the spec of the original component instead of a partial update.

Reply:
I think that's fair.
Why not just do:
```
  const resource =
    verb === 'update' ? { ...originalComponent, spec: newComponent.spec } : newComponent;
```

Reply:
Updated the code.